backend/src/api/blocks.ts

```diff
@@ -268,17 +268,19 @@ class Blocks {
         this.lastDifficultyAdjustmentTime = block.timestamp;
         this.currentDifficulty = block.difficulty;
 
-        const previousPeriodBlockHash = await bitcoinApi.$getBlockHash(blockHeightTip - heightDiff - 2016);
-        const previousPeriodBlock = await bitcoinApi.$getBlock(previousPeriodBlockHash);
-        this.previousDifficultyRetarget = (block.difficulty - previousPeriodBlock.difficulty) / previousPeriodBlock.difficulty * 100;
-        logger.debug(`Initial difficulty adjustment data set.`);
+        if (blockHeightTip - heightDiff - 2016 >= 0) {
```

Comment:
Is this new check to avoid computing the difficulty adjustment for the first 2016 blocks? If that's the case, it would be clearer to just have:
> `if (blockHeightTip > 2016)`

Reply:
Yes, mainly because $getBlockHash needs a positive integer, so i just copied that argument from there to compare it to 0, for clarity.

Is `blockHeightTip >= 2016` also correct?

Reply:
> Is blockHeightTip >= 2016 also correct?
Yes I think it's correct

Reply:
ok, changing this